File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSBooleanLiteralImpl.java
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import net.kenro.ji.jin.purescript.psi.PSBooleanLiteral;
import net.kenro.ji.jin.purescript.psi.PSValueRef;
import net.kenro.ji.jin.purescript.psi.PSVisitor;
import org.jetbrains.annotations.NotNull;

public class PSBooleanLiteralImpl extends PSPsiElement implements PSBooleanLiteral {

    public PSBooleanLiteralImpl(ASTNode node) {
        super(node);
    }

    public void accept(@NotNull PSVisitor visitor) {
        visitor.visitPSBooleanLiteral(this);
    }

    public void accept(@NotNull PsiElementVisitor visitor) {
        if (visitor instanceof PSVisitor) accept((PSVisitor) visitor);
        else super.accept(visitor);
    }





}

File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSBooleanLiteralImpl.kt
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import net.kenro.ji.jin.purescript.psi.PSBooleanLiteral;

public class PSBooleanLiteralImpl extends PSPsiElement implements PSBooleanLiteral {

    public PSBooleanLiteralImpl(final ASTNode node) {
        super(node);
    }

}
